src/Boundaries/Sediments/IronPhosphate.jl

```diff
@@ -0,0 +1,283 @@
+                                        FeS₂ = model.fields.FeS₂,
+                                        SO₄ = model.fields.SO₄,
+                                        TH₂S = model.fields.TH₂S,
+                                        CH₄ = model.fields.CH₄,
+                                        Gi = model.fields.Gi)
+
+@inline required_tracers(::IronPhosphate, bgc, tracers) = tracers[(:NO₃, :NH₄, :O₂, :DIC)] #, sinking_tracers(bgc)...)]
+@inline required_tendencies(::IronPhosphate, bgc, tracers) = tracers[(:NO₃, :NH₄, :O₂, :DIC)] # TODO add effluxes of PO4 and Fe once PISCES is working
+
+@inline bottom_index_array(sediment::IronPhosphate) = sediment.bottom_indices
+
+function _calculate_sediment_tendencies!(i, j, sediment::IronPhosphate, bgc, grid, advection, tracers, tendencies, sediment_tendencies, t)
+    k = bottom_index(i, j, sediment)
+    depth = @inbounds -znodes(grid, Center(), Center(), Center())[k]
+
+    Δz = zspacing(i, j, k, grid, Center(), Center(), Center())
+
+    @inbounds begin
+        oxygen_deposition = 0#oxygen_flux(i, j, k, grid, advection, bgc, tracers) * Δz
+
+        POC_deposition = 0#5e-9 #carbon_flux(i, j, k, grid, advection, bgc, tracers) * Δz
+
+        iron_deposition = 0#5e-10 #carbon_flux(i, j, k, grid, advection, bgc, tracers) * Δz * (1/106) * 0.1
+        
+        #(:O₂, :NH₄, :NO₃, :NO₂, :N₂, :TPO₄, :FeOHP, :Feᴵᴵ, :FeS₂, :SO₄, :TH₂S, :CH₄, :TCO₂, :Gi)
+        O₂ = sediment.fields.O₂[i, j, 1]
+        NH₄ = sediment.fields.NH₄[i, j, 1]
+        NO₃ = sediment.fields.NO₃[i, j, 1]
+        NO₂ = sediment.fields.NO₂[i, j, 1]
+        TPO₄ = sediment.fields.TPO₄[i, j, 1]
+        FeOHP = sediment.fields.FeOHP[i, j, 1]
+        Feᴵᴵ = sediment.fields.Feᴵᴵ[i, j, 1]
+        FeS₂ = sediment.fields.FeS₂[i, j, 1]
+        SO₄ = sediment.fields.SO₄[i, j, 1]
+        TH₂S = sediment.fields.TH₂S[i, j, 1]
+        CH₄ = sediment.fields.CH₄[i, j, 1]
+        Gi = sediment.fields.Gi[i, j, 1]
+
+        #println(FeOHP + FeS₂ + Feᴵᴵ) # iron is conserved
+        #println(2*FeS₂+SO₄+TH₂S) # sulfur is conserved
+        #saprintln(TPO₄)
+        #####
+        ##### RATES
+        #####
+
+        @inline K_O₂ = 1e-6 # M, Half–saturation constant for O2
+        @inline K_NO₃ = 10e-6 # M, Half–saturation constant for NO3
+        @inline K_NO₂ = 10e-6 # M, Half–saturation constant for NO2
+        @inline K_Fe = 0.028 # wt-%, Half–saturation constant for Fe
+        @inline K_SO₄ = 0.1e-6 # M, Half–saturation constant for SO4
+        @inline K_TPO₄ = 10e-6 # M, Half–saturation constant for TPO4
+        #(:O₂, :NH₄, :NO₃, :NO₂, :N₂, :TPO₄, :FeOHP, :Feᴵᴵ, :FeS₂, :SO₄, :TH₂S, :CH₄, :TCO₂, :Gi)
+         
+        @inline kGi = 0.016 # day⁻¹, Rate constant for G0 degradation, Dale et al
+        @inline fT = 1 # TODO temperature correction for rates
+        @inline fₒₓ = 10 # Enhancement factor for POM degradation by O2
+        @inline per_day_to_per_seconds = 1 / (60 * 60 * 24) 
```

Comment:
btw if you do `using Oceananigans.Units` then you can just use units like below instead of using this variable you can write `max(whatever)/day`